@3rdweb-sdk/react/components/connect-wallet/index.tsx

```diff
@@ -72,20 +75,28 @@ const connectorIdToImageUrl: Record<string, StaticImageData> = {
 
 export const ConnectWallet: React.FC<ButtonProps> = (buttonProps) => {
   const [connector, connect] = useConnect();
-  const { address, chainId, getNetworkMetadata } = useWeb3();
+  const { getNetworkMetadata } = useWeb3();
   const { isOpen, onOpen, onClose } = useDisclosure();
   const disconnect = useDisconnect();
   const disconnectFully = useDisconnect({ reconnectAfterGnosis: false });
   const [network, switchNetwork] = useNetwork();
+  const address = useAddress();
+  const chainId = useChainId();
 
   const { hasCopied, onCopy } = useClipboard(address || "");
 
+  const registerConnector = (_connector: string) => {
+    posthog.register({ connector: _connector });
+    posthog.capture("wallet_connected", { connector: _connector });
+  };
+
   function handleConnect(_connector: Connector<any, any>) {
     if (_connector.name.toLowerCase() === "magic") {
       onOpen();
     } else {
       connect(_connector);
     }
+    registerConnector(_connector.name);
```

Comment:
this means we'll `register` magic when the modal opens, not when it's actually connected